### PR TITLE
Improved SODA model

### PR DESCRIPTION
**Pull request by xseris**

### CODE REVIEW

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 1]
```diff
@@ -182,5 +186,37 @@ public String getContact_email() {
   public void setContact_email(String contact_email) {
     this.contact_email = contact_email;
   }
-  
+
+  public String getLens_view_type() {
+    return lens_view_type;
+  }
+
+  public void setLens_view_type(String lens_view_type) {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00117)
MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 2]
```diff
@@ -182,5 +186,37 @@ public String getContact_email() {
...
+    this.lens_view_type = lens_view_type;
+  }
+
+  public String getBlob_mime_type() {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 3]
```diff
@@ -182,5 +186,37 @@ public String getContact_email() {
...
+    return blob_mime_type;
+  }
+
+  public void setBlob_mime_type(String blob_mime_type) {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00117)
MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 4]
```diff
@@ -4,24 +4,28 @@
...
   String id;
-  String[] parent_fxf;
   String description;
   String attribution;
-  String attribution_link;
-  String contact_email;
   String type;
   String updatedAt;
   String createdAt;
   String metadata_updated_at;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "metadata_updated_at" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 5]
```diff
@@ -4,24 +4,28 @@
...
   String data_updated_at;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "data_updated_at" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 6]
```diff
@@ -4,24 +4,28 @@
...
   SodaPageViews page_views;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "page_views" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 7]
```diff
@@ -182,5 +186,37 @@ public String getContact_email() {
...
+
+  public String getBlob_mime_type() {
+    return blob_mime_type;
+  }
+
+  public void setBlob_mime_type(String blob_mime_type) {
+    this.blob_mime_type = blob_mime_type;
+  }
+
+  public boolean isHide_from_data_json() {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 8]
```diff
@@ -4,24 +4,28 @@
...
-  String attribution_link;
-  String contact_email;
   String type;
   String updatedAt;
   String createdAt;
   String metadata_updated_at;
   String data_updated_at;
   SodaPageViews page_views;
-  String[] columns_name;
   String[] columns_field_name;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "columns_field_name" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 9]
```diff
@@ -4,24 +4,28 @@
...
-  String[] columns_datatype;
+  String[] columns_name;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "columns_name" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 10]
```diff
@@ -4,24 +4,28 @@
...
   String[] columns_description;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "columns_description" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 11]
```diff
@@ -4,24 +4,28 @@
...
+  String[] columns_datatype;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "columns_datatype" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 12]
```diff
@@ -182,5 +186,37 @@ public String getContact_email() {
...
+
+  public void setBlob_mime_type(String blob_mime_type) {
+    this.blob_mime_type = blob_mime_type;
+  }
+
+  public boolean isHide_from_data_json() {
+    return hide_from_data_json;
+  }
+
+  public void setHide_from_data_json(boolean hide_from_data_json) {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00117)
MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 13]
```diff
@@ -4,24 +4,28 @@
...
   String metadata_updated_at;
   String data_updated_at;
   SodaPageViews page_views;
-  String[] columns_name;
   String[] columns_field_name;
-  String[] columns_datatype;
+  String[] columns_name;
   String[] columns_description;
+  String[] columns_datatype;
   SodaColumnFormat[] columns_format;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "columns_format" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 14]
```diff
@@ -4,24 +4,28 @@
...
-  long download_count;
+  String[] parent_fxf;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "parent_fxf" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 15]
```diff
@@ -4,24 +4,28 @@
...
   String provenance;
+  long download_count;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "download_count" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 16]
```diff
@@ -182,5 +186,37 @@ public String getContact_email() {
...
+
+  public boolean isHide_from_data_json() {
+    return hide_from_data_json;
+  }
+
+  public void setHide_from_data_json(boolean hide_from_data_json) {
+    this.hide_from_data_json = hide_from_data_json;
+  }
+
+  public String getPublication_date() {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 17]
```diff
@@ -4,24 +4,28 @@
...
-  String[] columns_datatype;
+  String[] columns_name;
   String[] columns_description;
+  String[] columns_datatype;
   SodaColumnFormat[] columns_format;
-  long download_count;
+  String[] parent_fxf;
   String provenance;
+  long download_count;
+  String lens_view_type;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "lens_view_type" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 18]
```diff
@@ -4,24 +4,28 @@
...
+  String blob_mime_type;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "blob_mime_type" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 19]
```diff
@@ -4,24 +4,28 @@
...
+  boolean hide_from_data_json;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "hide_from_data_json" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 20]
```diff
@@ -4,24 +4,28 @@
...
+  String attribution_link;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "attribution_link" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 21]
```diff
@@ -182,5 +186,37 @@ public String getContact_email() {
...
+
+  public void setHide_from_data_json(boolean hide_from_data_json) {
+    this.hide_from_data_json = hide_from_data_json;
+  }
+
+  public String getPublication_date() {
+    return publication_date;
+  }
+
+  public void setPublication_date(String publication_date) {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00117)
MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 22]
```diff
@@ -4,24 +4,28 @@
...
   SodaColumnFormat[] columns_format;
-  long download_count;
+  String[] parent_fxf;
   String provenance;
+  long download_count;
+  String lens_view_type;
+  String blob_mime_type;
+  boolean hide_from_data_json;
+  String attribution_link;
+  String contact_email;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "contact_email" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 23]
```diff
@@ -4,24 +4,28 @@
...
+  String publication_date;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this field "publication_date" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 24]
```diff
@@ -182,5 +186,37 @@ public String getContact_email() {
   public void setContact_email(String contact_email) {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00117)
MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java [hunk 25]
```diff
@@ -182,5 +186,37 @@ public String getContact_email() {
...
     this.contact_email = contact_email;
   }
-  
+
+  public String getLens_view_type() {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java [hunk 26]
```diff
@@ -50,4 +51,12 @@ public void setDomain_metadata(Map<String, String>[] domain_metadata) {
     this.domain_metadata = domain_metadata;
   }
 
+  public Map<String, String>[] getDomain_private_metadata() {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java [hunk 27]
```diff
@@ -9,6 +9,7 @@
   String domain_category;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java]: MINOR Rename this field "domain_category" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java [hunk 28]
```diff
@@ -9,6 +9,7 @@
...
   String[] domain_tags;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java]: MINOR Rename this field "domain_tags" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java [hunk 29]
```diff
@@ -50,4 +51,12 @@ public void setDomain_metadata(Map<String, String>[] domain_metadata) {
     this.domain_metadata = domain_metadata;
   }
 
+  public Map<String, String>[] getDomain_private_metadata() {
+    return domain_private_metadata;
+  }
+
+  public void setDomain_private_metadata(Map<String, String>[] domain_private_metadata) {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java]: MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00117)
MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java [hunk 30]
```diff
@@ -9,6 +9,7 @@
   String domain_category;
   String[] domain_tags;
   Map<String, String>[] domain_metadata;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java]: MINOR Rename this field "domain_metadata" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java [hunk 31]
```diff
@@ -9,6 +9,7 @@
...
+  Map<String, String>[] domain_private_metadata;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java]: MINOR Rename this field "domain_private_metadata" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResult.java [hunk 32]
```diff
@@ -55,4 +56,13 @@ public SodaOwner getOwner() {
   public void setOwner(SodaOwner owner) {
     this.owner = owner;
   }
+
+  public String getPreview_image_url() {
+    return preview_image_url;
+  }
+
+  public void setPreview_image_url(String preview_image_url) {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResult.java]: MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00117)
MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResult.java [hunk 33]
```diff
@@ -6,6 +6,7 @@
   SodaMetadata metadata;
   String permalink;
   String link;
+  String preview_image_url;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResult.java]: MINOR Rename this field "preview_image_url" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00116)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaResult.java [hunk 34]
```diff
@@ -55,4 +56,13 @@ public SodaOwner getOwner() {
   public void setOwner(SodaOwner owner) {
     this.owner = owner;
   }
+
+  public String getPreview_image_url() {
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaResult.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

**Comment by okkamit** [comment]
SonarQube analysis reported 84 issues
* ![MINOR][MINOR] 84 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [SodaClassification.java#L30](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L30): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [SodaClassification.java#L34](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L34): Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [SodaClassification.java#L34](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L34): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [SodaClassification.java#L38](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L38): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [SodaClassification.java#L42](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L42): Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [SodaClassification.java#L42](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L42): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [SodaClassification.java#L46](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L46): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [SodaClassification.java#L50](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L50): Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [SodaClassification.java#L50](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L50): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [SodaResource.java#L46](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java#L46): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00100)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by xseris** [comment]
Jenkins test it

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java [hunk 35]
```diff
@@ -50,4 +51,12 @@ public void setDomain_metadata(Map<String, String>[] domain_metadata) {
     this.domain_metadata = domain_metadata;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java]: MAJOR it.okkam.opendata.models.soda.SodaClassification.setDomain_metadata(Map[]) may expose internal representation by storing an externally mutable object into SodaClassification.domain_metadata [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP2)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java [hunk 36]
```diff
@@ -50,4 +51,12 @@ public void setDomain_metadata(Map<String, String>[] domain_metadata) {
...
   }
 
+  public Map<String, String>[] getDomain_private_metadata() {
+    return domain_private_metadata;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java]: MAJOR it.okkam.opendata.models.soda.SodaClassification.getDomain_private_metadata() may expose internal representation by returning SodaClassification.domain_private_metadata [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java [hunk 37]
```diff
@@ -50,4 +51,12 @@ public void setDomain_metadata(Map<String, String>[] domain_metadata) {
...
+  }
+
+  public void setDomain_private_metadata(Map<String, String>[] domain_private_metadata) {
+    this.domain_private_metadata = domain_private_metadata;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java]: MAJOR it.okkam.opendata.models.soda.SodaClassification.setDomain_private_metadata(Map[]) may expose internal representation by storing an externally mutable object into SodaClassification.domain_private_metadata [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP2)

File: soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java [hunk 38]
```diff
@@ -9,6 +9,7 @@
   String domain_category;
   String[] domain_tags;
   Map<String, String>[] domain_metadata;
+  Map<String, String>[] domain_private_metadata;
 
   public String[] getCategories() {
     return categories;
```
Comment (okkamit) [inline soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java]: MAJOR it.okkam.opendata.models.soda.SodaClassification.getCategories() may expose internal representation by returning SodaClassification.categories [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP)

**Comment by okkamit** [comment]
SonarQube analysis reported 106 issues
* ![MAJOR][MAJOR] 22 major
* ![MINOR][MINOR] 84 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SodaClassification.java#L19](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L19): it.okkam.opendata.models.soda.SodaClassification.setCategories(String[]) may expose internal representation by storing an externally mutable object into SodaClassification.categories [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP2)
1. ![MAJOR][MAJOR] [SodaClassification.java#L23](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L23): it.okkam.opendata.models.soda.SodaClassification.getTags() may expose internal representation by returning SodaClassification.tags [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP)
1. ![MAJOR][MAJOR] [SodaClassification.java#L27](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L27): it.okkam.opendata.models.soda.SodaClassification.setTags(String[]) may expose internal representation by storing an externally mutable object into SodaClassification.tags [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP2)
1. ![MAJOR][MAJOR] [SodaClassification.java#L39](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L39): it.okkam.opendata.models.soda.SodaClassification.getDomain_tags() may expose internal representation by returning SodaClassification.domain_tags [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP)
1. ![MAJOR][MAJOR] [SodaClassification.java#L43](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L43): it.okkam.opendata.models.soda.SodaClassification.setDomain_tags(String[]) may expose internal representation by storing an externally mutable object into SodaClassification.domain_tags [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP2)
1. ![MAJOR][MAJOR] [SodaClassification.java#L47](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaClassification.java#L47): it.okkam.opendata.models.soda.SodaClassification.getDomain_metadata() may expose internal representation by returning SodaClassification.domain_metadata [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP)
1. ![MAJOR][MAJOR] [SodaResource.java#L47](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java#L47): it.okkam.opendata.models.soda.SodaResource.getParent_fxf() may expose internal representation by returning SodaResource.parent_fxf [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP)
1. ![MAJOR][MAJOR] [SodaResource.java#L51](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java#L51): it.okkam.opendata.models.soda.SodaResource.setParent_fxf(String[]) may expose internal representation by storing an externally mutable object into SodaResource.parent_fxf [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP2)
1. ![MAJOR][MAJOR] [SodaResource.java#L119](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java#L119): it.okkam.opendata.models.soda.SodaResource.getColumns_name() may expose internal representation by returning SodaResource.columns_name [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP)
1. ![MAJOR][MAJOR] [SodaResource.java#L123](https://github.com/okkam-it/opendata-models/blob/26e0b0c4fc1f6f3beeb498e180f25df67b824d6a/soda/src/main/java/it/okkam/opendata/models/soda/SodaResource.java#L123): it.okkam.opendata.models.soda.SodaResource.setColumns_name(String[]) may expose internal representation by storing an externally mutable object into SodaResource.columns_name [rule](https://dev.okkam.it/sonar/coding_rules#rule_key=findbugs%3AEI_EXPOSE_REP2)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**